modal/_utils/function_utils.py

```diff
@@ -133,6 +133,9 @@ def __init__(
         elif f is None and user_cls:
             # "service function" for running all methods of a class
             self.function_name = f"{user_cls.__name__}.*"
+        elif f and user_cls:
```

Comment:
I don't think we should even be constructing FunctionInfo for methods (except @build ones, and we might want to deprecate those), but that's a separate matter.